packages/fxa-settings/src/components/Settings/PageTwoStepAuthentication/index.tsx

```diff
@@ -184,6 +184,12 @@ export const PageTwoStepAuthentication = (_: RouteComponentProps) => {
       GleanMetrics.accountPref.twoStepAuthEnterCodeView();
   }, [recoveryCodesAcknowledged, totpVerified]);
 
+  useEffect(() => {
+    showQrCode &&
+      totpInfo.result &&
+      GleanMetrics.accountPref.twoStepAuthQrView();
+  }, [showQrCode, totpInfo.result]);
```

Comment:
Looking at the returned content for when the QR code is displayed I think we should include this condition: https://github.com/mozilla/fxa/pull/17745/files?diff=unified&w=0#diff-f384404332bd3f4682466d1812169c5111dfa17b890f02d7d2e8287800363e26L206

Reply:
Fair call, updated!

Interestingly, I don't see tests that specifically check this state property changing.